webapp/graphite/render/datalib.py

```diff
@@ -108,22 +114,34 @@ def getInfo(self):
 
 @logtime()
 def _fetchData(pathExpr, startTime, endTime, now, requestContext, seriesList):
-  local = settings.REMOTE_PREFETCH_DATA or requestContext['localOnly']
+  result_queue = []
+  remote_done = False
+
+  if settings.REMOTE_PREFETCH_DATA:
+    prefetched = requestContext['prefetched'].get((startTime, endTime, now), None)
+    if prefetched is not None:
+      for result in prefetched[pathExpr]:
+        result_queue.append(result)
+        # Since we pre-fetched remote data only, now we can get local data only.
+        remote_done = True
+
+  local = remote_done or requestContext['localOnly']
```

Comment:
wrong indentation, fixed. We only set remote_done to False if we don't query with the same start/end/now.